### PR TITLE
Feat/delete rename subfolders

### PR DESCRIPTION
**Pull request by kwajiehao**
## Note
This PR is to be reviewed together with PR #[136](https://github.com/isomerpages/isomercms-backend/pull/136) on the backend, as that PR contains the necessary routes for renaming and deleting subfolders

## Overview

This PR introduces renaming and deleting functionality for subfolders. Renaming functionality is introduced by expanding the renaming logic inside `FolderModal` to include subfolders, and not just folders and resource categories. Deleting functionality is introduced by expanding the logic inside the delete handler in the `Folders` layout to account for subfolder `FolderContentItem`s in addition to pages.

Additionally, the `FolderModal` component was refactored to use `react-query`, in line with our recent PRs to refactor all data-fetching functionality to use `react-query`.

**Note:** the delete functionality for folder pages still does not work. It does not fall within the scope of this PR, and will be completed in a separate PR.

### CODE REVIEW

File: src/components/FolderModal.jsx [hunk 1]
```diff
@@ -13,23 +21,49 @@ import { errorToast } from '../utils/toasts';
...
+    return renameSubfolder(params)
+  }
+
+  const params = {
+    siteName,
+    categoryName: folderOrCategoryName,
+    newCategoryName: newDirectoryName,
+  }
+  return renameResourceCategory(params)
+}
```
Comment (alexanderleegs) [inline src/components/FolderModal.jsx:50]: We should slugify the new folder/subfolder/category names

**Review by alexanderleegs — CHANGES_REQUESTED** [review]
Left a comment regarding generation of folder names. Also, the page deletion warning modal and error toast still refer to subfolders, but we can fix that in a future PR, when we fix the deletion of pages.

File: src/components/FolderModal.jsx (hunk 1 above)
Comment (kwajiehao) [reply src/components/FolderModal.jsx]: resolved with [50b0b62](https://github.com/isomerpages/isomercms-frontend/pull/383/commits/50b0b625cdad0961e4c725471f97bd5f56e1136a)

**Comment by kwajiehao** [comment]
> Left a comment regarding generation of folder names. Also, the page deletion warning modal and error toast still refer to subfolders, but we can fix that in a future PR, when we fix the deletion of pages.

The mistaken error messages were a bug, I fixed them with this commit [b6d001c](https://github.com/isomerpages/isomercms-frontend/pull/383/commits/b6d001c8bc021134a921e2452a5810c7f5a60971)

**Review by alexanderleegs — APPROVED** [review]
lgtm!

**Pull request merged**